weierstrass-curves/src/weierstrass.rs

```diff
@@ -0,0 +1,475 @@
+#![allow(non_snake_case)]
+#![allow(dead_code)]
+#![allow(warnings, unused)]
+
+use hacspec_lib::*;
+
+#[derive(Copy, Clone)]
+pub struct WeiestrassCurve<T> {
+    a1: T,
+    a2: T,
+    a3: T,
+    a4: T,
+    a5: T,
+    a6: T,
+    bit_size_of_scalar_field: usize,
+}
+
+impl<T> WeiestrassCurve<T> {
+    /// Create a weierstrass curve
+    ///
+    /// The form of the weierstrass curve is:
+    /// Y^2+a_1XY+a_3Y = X^3+a_2X^2+a_4X+a_6
+    ///
+    /// # Arguments
+    ///
+    /// * `a1`
+    /// * `a2`
+    /// * `a3`
+    /// * `a4`
+    /// * `a5`
+    /// * `a6`
+    /// * `bit_size_of_scalar_field` - the size of the scalar field used for scalar multiplication
+    pub fn new(
+        a1: T,
+        a2: T,
+        a3: T,
+        a4: T,
+        a5: T,
+        a6: T,
+        bit_size_of_scaler_field: usize,
+    ) -> WeiestrassCurve<T> {
+        (WeiestrassCurve {
+            a1,
+            a2,
+            a3,
+            a4,
+            a5,
+            a6,
+            bit_size_of_scalar_field: bit_size_of_scaler_field,
+        })
+    }
+}
+
+/// Create a default instance of 'WeiestrassCurve<T>' with all values set to 0
+impl<T: Integer> Default for WeiestrassCurve<T> {
+    fn default() -> Self {
+        WeiestrassCurve {
+            a1: T::default(),
+            a2: T::default(),
+            a3: T::default(),
+            a4: T::default(),
+            a5: T::default(),
+            a6: T::default(),
+            bit_size_of_scalar_field: 0,
+        }
+    }
+}
+#[derive(Copy, Clone)]
+pub struct EllipticCurvePoint<T> {
+    x: T,
+    y: T,
+    isPointAtInfinity: bool,
+    curve: WeiestrassCurve<T>,
+}
+
```

Comment:
Have you considered raising an exception instead of using the point at infinity as a default value?

Reply:
Yes, that was the plan but i don't think that is hacspec complient, might be in v2?